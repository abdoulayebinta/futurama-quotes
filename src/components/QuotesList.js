import React, { Component } from 'react';
import Quote from './Quote';
import Pagination from './Pagination';


const QuotesList = (props) => {
  // Get current quotes
  const {
    currentPage, quotesPerPage, quotes, paginate,
  } = props;
  const indexOfLastQuote = currentPage * quotesPerPage;
  const indexOfFirstQuote = indexOfLastQuote - quotesPerPage;
  const currentQuotes = quotes.slice(indexOfFirstQuote, indexOfLastQuote);

  const quoteItems = currentQuotes.map(quote => (
    <Quote
      onQuoteSelect={props.onQuoteSelect}
      key={quote.quote}
      quote={quote}
    />
  ));


  return (
    <div>
      <div>
        <ul className="col-md-4 float-md-right list-group">
          {quoteItems}
        </ul>
      </div>
      <div>
        <Pagination
          quotesPerPage={quotesPerPage}
          totalQuotes={props.quotes.length}
          paginate={paginate}
        />
      </div>
    </div>
  );
};

export default QuotesList;
