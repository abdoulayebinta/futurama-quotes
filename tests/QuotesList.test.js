import React from 'react';
import renderer from 'react-test-renderer';
import QuotesList from '../src/components/QuotesList';

const listofQuotes = [ 
    { 'character': 'Fry', 
    'quote': "I haven't had time off since I was 21 through 24.", 
    'image': 'Fry.png' 
    },
    { 'character': 'Bender', 
    'quote': "Bender's quotes.", 
    'image': 'Bender.png' 
    }
]
    

describe('<QuotesList />', () => {
  test('Should not break if a quote is passed', () => {
    const component = renderer.create(<QuotesList  quotes={listofQuotes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(<QuotesList quotes={listofQuotes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
