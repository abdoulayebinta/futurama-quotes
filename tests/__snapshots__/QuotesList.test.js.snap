// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuotesList /> Should not break if a quote is passed 1`] = `
<div>
  <div>
    <ul
      className="col-md-4 float-md-right list-group"
    />
  </div>
  <div>
    <nav>
      <ul
        className="pagination"
      />
    </nav>
  </div>
</div>
`;

exports[`<QuotesList /> Should render correctly 1`] = `
<div>
  <div>
    <ul
      className="col-md-4 float-md-right list-group"
    />
  </div>
  <div>
    <nav>
      <ul
        className="pagination"
      />
    </nav>
  </div>
</div>
`;
